cli/injector/main.go

```diff
@@ -608,6 +608,9 @@ func watchTargetAndReinject(deadline time.Time, commandName string, pulseActiveD
 		select {
 		case sig := <-signals:
 			log.Infow("an exit signal has been received", "signal", sig.String())
+			// We've already consumed the signal from the channel, so our caller won't find it when checking after we return.
+			// Thus we need to put the signal back into the channel. We do it in a gothread in case we are blocked when writing to the channel
+			go func() { signals <- sig }()
```

Comment:
would love to see a unit test for this.

Reply:
I don't have a practical way to mock things such that we can test this literal line of code. In terms of testing this functionality of channels to prevent blocking in our caller, I've proven it out in the golang playground:
https://go.dev/play/p/UzCeUb2eh3C